lib/jobs/dell-wsman-add-hotspare-updatexml.js

```diff
@@ -0,0 +1,142 @@
+
+    function WsmanAddHotspareUpdateXmlJob(options, context, taskId) {
+        WsmanAddHotspareUpdateXmlJob.super_.call(this, logger, options, context, taskId);
+        assert.object(this.options);
+    }
+
+    util.inherits(WsmanAddHotspareUpdateXmlJob, BaseJob);
+
+    WsmanAddHotspareUpdateXmlJob.prototype._run = function() {
+        var self = this;
+        if(!self.options.driveId) {
+            throw new errors.NotFoundError('The driveId should not be empty.');
+        }
+        self.dell = configuration.get('dell');
+        if(!self.dell || !self.dell.shareFolder) {
+            throw new errors.NotFoundError('Share folder is not defined in smiConfig.json');
+        }
+        return self.updateXmlForRaid();
+    };
+
+    WsmanAddHotspareUpdateXmlJob.prototype.updateXmlForRaid = function() {
+        logger.info('Update xml for adding hotspare.');
+        var self = this;
+        var fileName = self.context.graphId + '.xml';
+        var client;
+        if(self.dell.shareFolder.shareType === 0) {
+            client = new NfsClient(
+                self.dell.shareFolder.address,
+                self.dell.shareFolder.shareName,
+                self.context.mountDir
+            );
+        } else if(self.dell.shareFolder.shareType === 2) {
+            client = new Smb2Client(
+                self.dell.shareFolder.address,
+                self.dell.shareFolder.shareName,
+                self.dell.shareFolder.username,
+                self.dell.shareFolder.password
+            );
+        } else {
+            throw new Error('Invalid shareType in smiConfig.json. The shareType should be 0 or 2.');
+        }
+        return client.readFile(fileName)
+        .then(function(data) {
+            return self.updateXml(data);
+        })
+        .then(function(doc) {
+            return client.writeFile(fileName, doc);
+        })
+        .then(function() {
+            self._done();
+        })
+        .catch(function(err) {
+            logger.error('An error occurred while updating component xml.', { error: err });
+            self._done(err);
+        });
+    };
+
+    WsmanAddHotspareUpdateXmlJob.prototype.updateXml = function(data) {
+        var self = this;
+        var xmlFile = String.fromCharCode.apply(null, new Uint16Array(data));
+        var doc = new xmldom().parseFromString(xmlFile, 'application/xml');
+        var components = doc.getElementsByTagName('Component');
+        if(self.options.hotspareType === 'dhs') {
+            //Dedicated hotspare should be added into corresponding volume
+            if(!self.options.volumeId) {
+                throw new errors.NotFoundError('The volumeId should not be empty.');
+            }
+            for(var i = 0; i < components.length; i++) { //jshint ignore:line
+                var volumeFQDD = components[i].getAttribute('FQDD');
+                if(volumeFQDD === self.options.volumeId) {
```

Comment:
Lines inside if are difficult to understand, you could put them into a function.